### PR TITLE
update stack and eck operator version

### PR DESCRIPTION
**Pull request by kvalliyurnatt**
Updating the stack versions in the release branch

### CODE REVIEW

**Review by naemono — CHANGES_REQUESTED** [review]
Also, all of the `deploy/eck-stack/*Chart.yaml` don't appear to be updated.  Check https://github.com/elastic/cloud-on-k8s/commit/9e500fff4f64d53c2234c8109199de68016dad07

File: .buildkite/e2e/nightly-main-matrix.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@
     - E2E_STACK_VERSION: "8.11.4"
     - E2E_STACK_VERSION: "8.12.2"
     # current stack version 8.13.x is tested in all other tests no need to test it again
-    - E2E_STACK_VERSION: "8.14.0-SNAPSHOT"
```
[the remark this answers is not in the document]
Comment (kvalliyurnatt) [reply .buildkite/e2e/nightly-main-matrix.yaml]: Thanks, I missed this completely. Updated.

File: hack/operatorhub/config.yaml [hunk 2]
```diff
@@ -41,7 +41,7 @@ packages:
   - outputPath: certified-operators
     packageName: elasticsearch-eck-operator-certified
     distributionChannel: certified-operators
-    minSupportedOpenshiftVersion: v4.11
+    minSupportedOpenshiftVersion: v4.12
```
Comment (naemono) [inline hack/operatorhub/config.yaml:44]: 👍 4.11 is EOL.

File: hack/upgrade-test-harness/conf.yaml [hunk 3]
```diff
@@ -42,5 +42,5 @@ testParams:
     operatorVersion: 2.12.1
     stackVersion: 8.13.2
   - name: upcoming
-    operatorVersion: 2.13.0-SNAPSHOT
-    stackVersion: 8.14.0-SNAPSHOT
+    operatorVersion: 2.13.0
+    stackVersion: 8.14.0
```
Comment (naemono) [inline hack/upgrade-test-harness/conf.yaml:46]: I don't think you want to bump this, as you won't be able to run the upgrade test harness, as 8.14 hasn't been released yet...

Comment (kvalliyurnatt) [reply hack/upgrade-test-harness/conf.yaml]: aah thanks!! fixed it.

**Pull request merged**